ansible/master_cfg/letsencrypt.sh

```diff
@@ -1,9 +1,22 @@
 #!/bin/bash
 
+# Updates on 2018-08-13 by ascheel to gracefully handle autorenewal
+# Add --keep-until-expiring, --post-hook "systemctl restart nginx" to letsencrypt-auto
+# Remove --renew-by-default aka --force-renewal from letsencrypt-auto
+# Made /etc/nginx/tls/server.{crt,key} symlinks to /etc/letsencrypt/live/jenkins.open-scap.org/{fullchain.pem,privkey.pem}
+# Updated crontab to execute every day, not once a month
+# Added logging of command output to /var/log/letsencrypt-renewal.log
+# Logs for renewal failures are in /var/logs/letsencrypt/letsencrypt.log* and are timestamped
+
+log="/var/log/letsencrypt-renewal.log"
+
+echo "=====BEGIN RENEWAL $(date)=====" >> $log
+
 # TODO: Change the email to something that makes more sense
-/root/letsencrypt/letsencrypt-auto certonly --webroot -w /letsencrypt_public_html/ -d jenkins.open-scap.org -m mpreisle@redhat.com --agree-tos -n --renew-by-default
+/root/letsencrypt/letsencrypt-auto certonly --keep-until-expiring --webroot -w /letsencrypt_public_html/ -d jenkins.open-scap.org -m mpreisle@redhat.com --agree-tos -n --post-hook "systemctl restart nginx" >> $log 2>&1
+
+ret=$?
+echo "letsencrypt-auto return code: $ret" >> $log 2>&1
+echo "=====END RENEWAL=====" >> $log
+exit $ret
 
-# deploy to nginx
-cp -L /etc/letsencrypt/live/jenkins.open-scap.org/fullchain.pem /etc/nginx/tls/server.crt
-cp -L /etc/letsencrypt/live/jenkins.open-scap.org/privkey.pem /etc/nginx/tls/server.key
```

Comment:
these are no longer necessary?

Reply:
Correct, per [this comment](https://github.com/OpenSCAP/jenkins/pull/54/files#diff-77e61d7d2477f13f7f7fd120b9ecbfecR6), I've made them symbolic links; since we are fully specifying one domain, the path should not change (and if we did, under the old script we'd have had problems as well). Further, error cases should be handled nicely in `letsencrypt-auto` so only copying these on success shouldn't be needed.

Reply:
One could call symbolic links "continual deployment". :)

Reply:
Well that assumes that the links are already there, right? So if this master fails or we have to move it (to wedos) we will have to redo the symlinks. I think it would be better if this script made sure the links are there.

Reply:
Ah okay, fair catch.

Should I add that to this script or to the Ansible [playbooks](https://github.com/OpenSCAP/jenkins/blob/master/ansible/master.yml#L291-L295)?

Reply:
Your call, I think playbooks probably make more sense